apple/swift-docc-render
# Improved theme settings support

Bug/issue #, if applicable: 97040358

## Summary

This PR updates the existing support for the `theme-settings.json` file to be more fully featured and capable. This is the Swift-DocC-Render implementation support for my recent forum proposal titled [
Customizing the look and feel of Swift-DocC-Render](https://forums.swift.org/t/customizing-the-look-and-feel-of-swift-docc-render/58858).

### Major Changes:
* removed default `theme-settings.json` file since it only existed for pseudo documentation purposes (full OpenAPI spec will be created for Swift-DocC as well as conceptual docs)
* fixed bugs with overwritten colors not working right when switching between Light/Dark/Auto modes
* added support for global border-radius override
* added support for overriding content and code font stacks
* added support for overriding builtin SVG icons
* added support for overriding component specific borders of the following elements
  + asides
  + badges
  + buttons
  + code listings
* added some new colors to make it easier to customize the follow elements:
  + buttons
  + documentation intro
  + tutorials overview page
* added support for customizing outer border radius of step elements
* all supported properties of the `theme-settings.json` file should be specified [here](https://github.com/mportiz08/swift-docc/blob/document-theme-settings-api/Sources/SwiftDocC/SwiftDocC.docc/Resources/ThemeSettings.spec.json) (will open PR on Swift-DocC for this soon)

More documentation PRs will be created for the swift-docc repo next to guide users on how to utilize this file.

## Dependencies

https://github.com/apple/swift-docc/pull/339 (already merged to Swift-DocC main)

## Testing

Steps:
1. From this branch, run `npm run build` to build this branch of swift-docc-render
2. Checkout [swift-docc](https://github.com/apple/swift-docc) and pull the latest main branch
3. Extract the example `theme-settings.json` file and custom `article.svg` icon from [files.zip](https://github.com/apple/swift-docc-render/files/9255665/files.zip) and copy them into the top level of the documentation catalog for SwiftDocC at `Sources/SwiftDocC/SwiftDocC.docc`
4. From the swift-docc repo, run `env DOCC_HTML_DIR=/path/to/swift-docc-render/dist bin/preview-docs SwiftDocC`
5. Open http://localhost:8000/documentation/swiftdocc and check that the theme settings are applied as expected
6. Play around with your own settings and check that there are no regressions if there are no theme settings provided

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
